src/config.h

```diff
@@ -391,4 +391,12 @@ void setcpuaffinity(const char *cpulist);
 #define ATTRIBUTE_TARGET_AVX512
 #endif
 
+#if defined(__linux__) &&                                                               \
+    ((defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 3))) || \
+     (defined(__clang__) && (__clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 4))))
```

Comment:
Do we expect valkey to be compiled with gcc4 and clang3 which are more than a decade old? If not, perhaps we can this simple: (clang also defines __GNUC__)

```
#if defined(__linux__) && (defined(__GNUC__) 
```

Reply:
Better safe. We can ignore 4.x to make the check simpler.

    #if defined(__linux__) && defined(__GNUC__) && __GNUC__ > 4

Reply:
Clang defines `__GNUC__` as 4 regardless of the actual Clang version, so this check fails with clang.

Reply:
So we need GNUC > 4 or clang_major > 3 then :)

Btw, if a macro isn't defined, for expample `__clang_major__`, don't you get a warning for `defined(__clang_major) && __clang_major__ > 3` when compiling with `-Wundef`?

Reply:
> So we need GNUC > 4 or clang_major > 3 then :)

yeah. But they so old though :) 

> Btw, if a macro isn't defined, for expample __clang_major__, don't you get a warning for defined(__clang_major) && __clang_major__ > 3 when compiling with -Wundef?

Looks like not if proceeded by `#if defined`. This seems to be build with `gcc main.c -Wundef`: 

```
#include <stdio.h>
int main(void) {
#if defined(__clang_major) && __clang_major__ > 3
    printf("The value of __clang__major__ is: %d\n", __clang_major__);
#endif
    return 0;
}
```
This one throws a warning: 

```
#include <stdio.h>
int main(void) {
#if __clang_major__ > 3
    printf("The value of __clang__major__ is: %d\n", __clang_major__);
#endif
    return 0;
}
```

Reply:
Interesting. Thanks, I learned something.

Reply:
FYI, the `__builtin_cpu_init()` function was introduced in Clang with version 6.0, released in 2018. and in GCC with version 4.8, released in 2013.